packages/next/build/index.ts

```diff
@@ -260,6 +261,7 @@ export default async function build(dir: string, conf = null): Promise<void> {
     path.join(distDir, ROUTES_MANIFEST),
     JSON.stringify({
       version: 1,
+      static404: !!(!pageKeys.includes('/404') || config.experimental.pages404),
```

Comment:
This seems incorrect, at this point in time `config.experimental.pages404` always has to be enabled when setting static404. Also 404.js could be not static, eg through getServerProps/getInitialProps, but it would still need the correct route in now/next.

Reply:
This is meant as a hint that if a `404.html` is present the config allows it to be used as a static 404 page, we could re-write to the `routes-manifest.json` with an explicit value but it shouldn't really be needed since this is just telling us whether their config allows using a `404.html` if present